### PR TITLE
Added beginning of CONTRIBUTING guide

### PR DESCRIPTION
**Pull request by IanLee1521**
Moved from: https://github.com/LLNL/llnl.github.io/pull/22

cc/ @LLNL/admins

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@IanLee1521 - I think this can be merged as is. The rest of the guide would be specific to the particular workflow

**Review by davidbeckingsale — APPROVED** [review]

File: CONTRIBUTING.md
Comment (dinatale2) [inline CONTRIBUTING.md]: `We encourage you to read this project's CONTRIBUTING policy (you are here)...` could be `We encourage you to read through this document...` The person reading this likely knows where they are already.

Also, comma should be before the `and` in `our CONTRIBUTING guidelines, LICENSE, and README` and you can drop `its`.

**Review by dinatale2 — COMMENTED** [review]
@IanLee1521 @davidbeckingsale Apologies for the random chime in, but it may be worth adding information regarding best practices. Some suggestions are:
- Strongly encourage that a project should have some form of continuous integration
- Assuming continuous integration, explicitly state all tests must pass and new tests must added if necessary
- Explicitly state that approval from maintainers is required before change is merged
- Encourage future maintainers to spell out their `Signed-off-by` policy or some form of policy to allow distribution of the change(s)
- If necessary, any tweaks to a license should be explained
- Might also be appropriate to encourage a project to adopt coding style (depends really on the size of the project for this)
- Commit message format requirements (ties into the `Signed-off-by` bullet above)

It is also possible that it is not appropriate to put those in the CONTRIBUTING.md of this repo, but maybe a guidelines type of document perhaps? Or maybe as giant comment blocks in this template?

I've helped develop ZFS on Linux's contributing guidelines (found [here](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md)). I'm also helping mpiFileUtils get a preliminary one up (found in a PR [here](https://github.com/hpc/mpifileutils/pull/77)).

**Comment by IanLee1521** [comment]
@dinatale2 , the question then becomes should that be embedded into the CONTRIBUTING document, or more into a "best practices" section?

As a lot of those suggestions are more likely to vary from project to project, I'm hesitant to say they should all be in the template CONTRIBUTING file.. Thoughts?

**Comment by dinatale2** [comment]
@IanLee1521 Maybe a suggested best practices document that exists outside of this document?

**Comment by IanLee1521** [comment]
I'm thinking of a new directory in this repo to contain topics like: "Working with Git" (recommended workflow, e.g. git-flow, e.g. #4), "Creating Releases" (signing commits / tags, creating releases, etc), "Continuous Integration" (quick start for Travis.ci, maybe with some pointers to example repos in the org for a few different programming languages?), etc

**Comment by IanLee1521** [comment]
There you go @dinatale2 !

**Pull request merged**
